src/controllers/bookController.js

```diff
@@ -0,0 +1,195 @@
+const Book = require('../models/Book');
+const { NotFoundError, BadRequestError } = require('../errors');
+const { StatusCodes } = require('http-status-codes');
+
+const getAllBooks = async (req, res, next) => {
+  try {
+    const {
+      query, //general search
+      title,
+      author,
+      isbn,
+      limit = 12,
+      skip = 0,
+      ageCategory,
+      condition,
+      genre,
+      coverType,
+      sort = '-createdAt',
+      userId,
+    } = req.query;
+
+    const queryObj = {};
+
+    //filter by user
+    if (userId) {
+      queryObj.createdBy = userId;
+    }
+
+    //General search by "query"
+    if (query) {
+      queryObj.$or = [
+        { title: { $regex: query, $options: 'i' } },
+        { author: { $regex: query, $options: 'i' } },
+        { isbn10: { $regex: query, $options: 'i' } },
+        { isbn13: { $regex: query, $options: 'i' } },
+      ];
+    }
+
```

Comment:
Nothing to fix, I just want you to know that multiple or/and queries are inefficient and you would add indexing in mongodb to fix this.